docs/source/user-guide/tuning.md

```diff
@@ -17,18 +17,94 @@ specific language governing permissions and limitations
+Comet to share an off-heap memory pool with Spark, reducing the overall memory overhead. The size of the pool is
+specified by `spark.memory.offHeap.size`. For more details about Spark off-heap memory mode, please refer to
+Spark documentation: https://spark.apache.org/docs/latest/configuration.html.
+
+### Configuring Comet Memory in On-Heap Mode
+
+When running in on-heap mode, Comet memory can be allocated by setting `spark.comet.memoryOverhead`. If this setting
+is not provided, it will be calculated by multiplying the current Spark executor memory by
+`spark.comet.memory.overhead.factor` (default value is `0.2`) which may or may not result in enough memory for 
+Comet to operate. It is not recommended to rely on this behavior. It is better to specify `spark.comet.memoryOverhead`
+explicitly.
+
+Comet supports native shuffle and columnar shuffle (these terms are explained in the [shuffle] section below). 
+In on-heap mode, columnar shuffle memory must be separately allocated using `spark.comet.columnar.shuffle.memorySize`. 
+If this setting is not provided, it will be calculated by multiplying `spark.comet.memoryOverhead` by
+`spark.comet.columnar.shuffle.memory.factor` (default value is `1.0`). If a shuffle exceeds this amount of memory 
+then the query will fail.
+
+[shuffle]: #shuffle
+
+### Determining How Much Memory to Allocate
+
+Generally, increasing the amount of memory allocated to Comet will improve query performance by reducing the
+amount of time spent spilling to disk, especially for aggregate, join, and shuffle operations. Allocating insufficient
+memory can result in out-of-memory errors. This is no different from allocating memory in Spark and the amount of
+memory will vary for different workloads, so some experimentation will be required.
+
+Here is a real-world example, based on running benchmarks derived from TPC-H.
+
+**TODO: this section is a work-in-progress**
+
+The following table shows performance of Spark compared to Comet in both Off-Heap and On-Heap modes. The table shows
+total query time for TPC-H @ 100GB. Smaller is better.
+
+| Total Executor Memory (GB) | Spark | Comet Off-Heap | Comet On-Heap |
+| -------------------------- | ----- | -------------- | ------------- |
+| 1                          | OOM   | OOM            | OOM           |
+| 2                          | OOM   | OOM            | OOM           |
+| 3                          | 744   | OOM            | OOM           |
+| 4                          | 739   | OOM            | OOM           |
+| 5                          | 681   | 342            | OOM           |
+| 6                          | 665   |                | 344           |
+| 7                          | 657   |                | 340           |
+| 8                          | 632   | 295            | 334           |
+| 9                          | 623   |                |               |
+| 10                         | 622   |                |               |
```

Comment:
Are these tests ran using a single CPU core in local mode? I think we need to provide more information about the Spark and hardware configurations we used when running these tests.

Reply:
This is with 8 cores single executor - I have removed this table because it was overkill and have now just summarized the key findings